src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -28,7 +28,11 @@ public static class TagHelperBlockRewriter
             var attributes = GetTagAttributes(tagName, validStructure, tag, descriptors, errorSink);
             var selfClosing = IsSelfClosing(tag);
 
-            return new TagHelperBlockBuilder(tagName, selfClosing, start, attributes, descriptors);
+            // All descriptors for an element should share a common prefix, grab the first.
+            var prefix = descriptors.First().Prefix;
+            var tagHelperTagName = tagName.Substring(prefix.Length);
```

Comment:
why does this code exist?  the descriptor should contain all `string`s of interest; no `Substring` required